### PR TITLE
Bringing inflation in line with upstream

### PR DESCRIPTION
**Pull request by ayrton04**
@skaynama @p6chen 

(FYI @jasonimercer)

Note that I am merging this into the indigo-devel-cpr-costmaps branch.

### CODE REVIEW

File: costmap_2d/plugins/inflation_layer.cpp [hunk 1]
```diff
@@ -66,40 +60,38 @@ InflationLayer::InflationLayer()
   , seen_(NULL)
   , cached_costs_(NULL)
   , cached_distances_(NULL)
+  , last_min_x_(-std::numeric_limits<float>::max())
```
Comment (skaynama) [inline costmap_2d/plugins/inflation_layer.cpp]: if ```last_min_x_``` is of type double shouldn't this be a double max limit?

File: costmap_2d/plugins/inflation_layer.cpp [hunk 2]
```diff
@@ -146,23 +127,43 @@ void InflationLayer::updateBounds(double robot_x, double robot_y, double robot_y
 {
   if (need_reinflation_)
   {
+    last_min_x_ = *min_x;
+    last_min_y_ = *min_y;
+    last_max_x_ = *max_x;
+    last_max_y_ = *max_y;
+    // For some reason when I make these -<double>::max() it does not
+    // work with Costmap2D::worldToMapEnforceBounds(), so I'm using
+    // -<float>::max() instead.
```
Comment (skaynama) [inline costmap_2d/plugins/inflation_layer.cpp]: Ah never mind! Odd.

**Comment by skaynama** [comment]
LGTM.

File: costmap_2d/plugins/inflation_layer.cpp (hunk 1 above)
Comment (ayrton04) [inline costmap_2d/plugins/inflation_layer.cpp]: I found code elsewhere in navigation where the author complained that using -std::numeric_limits<double>::max() was causing for them.

File: costmap_2d/plugins/inflation_layer.cpp (hunk 2 above)
Comment (ayrton04) [inline costmap_2d/plugins/inflation_layer.cpp]: Ah, there it is.

**Comment by ayrton04** [comment]
@p6chen, does this look OK to you?

**Comment by p6chen** [comment]
Seems ok to me, though I am not super familiar with this code base.

**Comment by ayrton04** [comment]
There will be another PR when this branch goes back into indigo-devel, so @jasonimercer will get a second look at it then. Thanks all!

**Pull request merged**